lib/sharing/SharePlugin.js

```diff
@@ -43,12 +43,19 @@ class SharePlugin {
 			},
 			item => item
 		);
+		validateShareOptions(sharedOptions);
 		/** @type {Record<string, ConsumesConfig>[]} */
 		const consumes = sharedOptions.map(([key, options]) => ({
 			[key]: {
 				import: options.import,
 				shareKey: options.shareKey || key,
 				shareScope: options.shareScope,
+				exclusionCriteria: options.exclusionCriteria
+					? {
+							...options.exclusionCriteria,
+							fallbackVersion: options.version ? options.version : undefined
+					  }
+					: undefined,
```

Comment:
Pass in `version` as `fallbackVersion` to exclusionCriteria. As far a consumed module is concerned, the `version` property represents the version of the `fallback`.